### ISSUE TITLE
Critical error at checkout after 2.4.0 update

### ISSUE THREAD

**Issue by JoshuaGoode** [krokedil/klarna-payments-for-woocommerce#220]
***Issue***
Critical error at checkout after 2.4.0 update

***Context***
We've received multiple reports on WordPress.com regarding users getting a critical error on their checkout pages after 2.4.0.

Due to the nature of our environment, it's not possible to roll this back and it doesn't appear some of the suggested workarounds are working. 

Additionally, it appears this is making it on to your forums at https://wordpress.org/support/topic/error-en-plugin-klarna-payments-for-woocommerce/

***Error***

```
PHP Fatal error:  Uncaught TypeError: Argument 1 passed to KP_IFrame::__construct() must be of the type array, bool given, called in /wordpress/plugins/klarna-payments-for-woocommerce/2.4.0/classes/requests/class-kp-requests.php on line 48 and defined in /wordpress/plugins/klarna-payments-for-woocommerce/2.4.0/classes/requests/helpers/class-kp-iframe.php:56
Stack trace:
#0 /wordpress/plugins/klarna-payments-for-woocommerce/2.4.0/classes/requests/class-kp-requests.php(48): KP_IFrame->__construct(false)
#1 /wordpress/plugins/klarna-payments-for-woocommerce/2.4.0/classes/requests/class-kp-requests.php(25): KP_Requests->set_environment_variables()
#2 /wordpress/plugins/klarna-payments-for-woocommerce/2.4.0/includes/kp-functions.php(37): KP_Requests->__construct()
#3 /wordpress/plugins/klarna-payments-for-woocommerce/2.4.0/classes/class-kp-checkout.php(34): kp_maybe_create_session_cart('GB')
#4 /wordpress/core/5.7/wp-includes/class-wp-hook.php(292): KP_Checkout->update_klarna_session(Object(WC_Cart))
#5 /wordpress/core/5.7/wp-includes/class-wp-hook.php(316): WP_Hook->apply_filters(NULL, Array)
#6 /wordpress/core/5.7/wp-includes/plugin.php(484): WP_Hook->do_action(Array)
#7 /wordpress/plugins/woocommerce/5.1.0/includes/class-wc-cart.php(1371): do_action('woocommerce_aft...', Object(WC_Cart))
#8 /wordpress/plugins/woocommerce/5.1.0/includes/shortcodes/class-wc-shortcode-checkout.php(275): WC_Cart->calculate_totals()
#9 /wordpress/plugins/woocommerce/5.1.0/includes/shortcodes/class-wc-shortcode-checkout.php(67): WC_Shortcode_Checkout::checkout()
#10 /wordpress/plugins/woocommerce/5.1.0/includes/class-wc-shortcodes.php(72): WC_Shortcode_Checkout::output('')
#11 /wordpress/plugins/woocommerce/5.1.0/includes/class-wc-shortcodes.php(95): WC_Shortcodes::shortcode_wrapper(Array, '')
#12 /wordpress/core/5.7/wp-includes/shortcodes.php(343): WC_Shortcodes::checkout('', '', 'woocommerce_che...')
#13 [internal function]: do_shortcode_tag(Array)
#14 /wordpress/core/5.7/wp-includes/shortcodes.php(218): preg_replace_callback('/\\[(\\[?)(woocom...', 'do_shortcode_ta...', '\n[woocommerce_c...')
#15 /wordpress/core/5.7/wp-includes/class-wp-hook.php(292): do_shortcode('\n[woocommerce_c...')
#16 /wordpress/core/5.7/wp-includes/plugin.php(212): WP_Hook->apply_filters('\n[woocommerce_c...', Array)
#17 /wordpress/core/5.7/wp-includes/post-template.php(253): apply_filters('the_content', '<!-- wp:paragra...')
#18 /wordpress/themes/twentytwenty/1.7/template-parts/content.php(36): the_content('Continue readin...')
#19 /wordpress/core/5.7/wp-includes/template.php(732): require('/wordpress/them...')
#20 /wordpress/core/5.7/wp-includes/template.php(676): load_template('/wordpress/them...', false, Array)
#21 /wordpress/core/5.7/wp-includes/general-template.php(204): locate_template(Array, true, false, Array)
#22 /wordpress/themes/twentytwenty/1.7/singular.php(24): get_template_part('template-parts/...', 'page')
#23 /wordpress/core/5.7/wp-includes/template-loader.php(106): include('/wordpress/them...')
#24 /wordpress/core/5.7/wp-blog-header.php(19): require_once('/wordpress/core...')
#25 /wordpress/core/5.7/index.php(17): require('/wordpress/core...')
#26 {main}
  thrown in /wordpress/plugins/klarna-payments-for-woocommerce/2.4.0/classes/requests/helpers/class-kp-iframe.php on line 56
  ```

**Comment by NiklasHogefjord** [comment]
Thanks for the report @JoshuaGoode,
This could absolutely be handled better in the plugin. However, I think the actual issue could be solved by the user navigating to the Klarna Payments settings page and then save the settings.

We will get a fix for this released during Thursday (CET).

**Comment by MichaelBengtsson** [comment]
Fixed in 3a6dabaab3adfec66239d5a6a1a3a5706e68bd0c. A release will be coming out shortly @JoshuaGoode
